lib/rules/base.js

```diff
@@ -30,7 +19,13 @@ module.exports = function(options) {
     this._log = log;
     this.ruleName = ruleName;
     this.severity = defaultSeverity;
+    // To support DOM-scoped configuration instructions, we need to maintain
+    // a stack of our configurations so we can restore the previous one when
+    // the current one goes out of scope. The current one is duplicated in
+    // this.config so the rule implementations don't need to worry about the
+    // fact that there is a stack.
     this.config = this.parseConfig(config);
+    this.configStack = [this.config];
```

Comment:
I was thinking that we don't want it to be public -- rules shouldn't be concerned about *how* their configuration is specified, just *what* the configuration is. My goal was to make that all a black box that resolves to a single `this.config` at any given time. So, unless you disagree I'll underscore `configStack`.